service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ValidateOrderController.java

```diff
@@ -43,11 +41,9 @@ public AboutToStartOrSubmitCallbackResponse handleMidEventValidateDateOfIssue(
         List<String> errors = validateGroupService.validateGroup(caseData, DateOfIssueGroup.class);
 
         if (GeneratedOrderSubtype.INTERIM.equals(caseData.getOrderTypeAndDocument().getSubtype())) {
-            if (caseData.getInterimEndDate() != null) { // TODO: IF To be removed after FPLA-2202 goes live
-                errors.addAll(validateGroupService.validateGroup(
-                    caseData.getInterimEndDate(), InterimEndDateGroup.class
-                ));
-            }
+            errors.addAll(validateGroupService.validateGroup(
```

Comment:
Do you need the condition that was removed from the other method?
```java
        if (interimEndDate.getType() == NAMED_DATE) {	
```
although maybe it is now NAMED_DATE and SPECIFIC_TIME_NAMED_DATE.
Cannot remember how nulls interact with the validation

Reply:
InterimEndDate uses @Future validators

https://docs.oracle.com/javaee/6/api/javax/validation/constraints/Future.html

nulls elements are considered valid.

Reply:
Missed the null elements part...
I need more sleep 🙂

Reply:
I've added some more tests that actually cover the case for time on the top of dates to make it more explicit. Doing that I probably figured out why those tests were not there 😄  Basically the LocalValidatorFactoryBean by default takes its own clock provider. That caused to have the test clock to be at fixed time, and the validator with the standard one. That meant flakiness if the test would run let say at minute 11:21, priming the data at that instant, and then running the validation at 11:22.

Reply:
I love time, always a joy to test 😅